demo/cpp/inout_transform.h

```diff
@@ -0,0 +1,87 @@
+#ifndef INOUT_TRANSFORM
+#define INOUT_TRANSFORM
+
+#include <iostream>
+#include <vector>
+#include <opencv2/core/core.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include "data.h"
+
+/*
+according to DCNv2 config file input image must pass following transformation:
+
+dict(
+    img_scale=(1333, 800),
+    transforms=[
+        dict(type='Resize', keep_ratio=True),
+        dict(type='Normalize', **img_norm_cfg),
+        dict(type='Pad', size_divisor=32),
+    ])
+normalize using mean=[123.675, 116.28, 103.53] and std=[58.395, 57.12, 57.375]
+*/
+
+cv::Size g_input_size(1333, 800);
+cv::Scalar g_mean_RGB{123.675f, 116.28f, 103.53f};
+cv::Scalar g_std_RGB{58.395f, 57.12f, 57.375f};
+unsigned g_pad_divisor = 32;
+
+
+class ImageTransformer {
+  public:
+    ImageTransformer(cv::Size const& input_layer_size):m_input_layer_size(input_layer_size) {
+        int pad_h = (int) (ceil(g_input_size.height / (float) g_pad_divisor)) * g_pad_divisor;
+        int pad_w = (int) (ceil(g_input_size.width  / (float) g_pad_divisor)) * g_pad_divisor;
+        m_padded_size = cv::Size(pad_w, pad_h);
+    }
+    
+    void transform_input_image(cv::Mat const& input_image, cv::Mat& output_image) const {
+        //  1. To RGB, FP32
+        cv::cvtColor(input_image, output_image, CV_BGR2RGB);
+        output_image.convertTo(output_image, CV_32FC3);
+        //  2. Resize to config size
+        cv::resize(output_image, output_image, g_input_size);
+        //  3. Normalize    
+        cv::Mat rgb[3];
+        cv::split(output_image, rgb);
+        for (size_t i=0; i<3; i++) {
+            rgb[i] = (rgb[i] - g_mean_RGB[i]) / g_std_RGB[i];
+        }
+        cv::merge(rgb, 3, output_image);
```

Comment:
```
output_image -= g_mean_RGB;
cv::divide(output_image, g_std_RGB, output_image);
```
would be better.

Reply:
I've tested it degrades processing speed a bit. Can do it in your code as you want, but in my prod will keep my variant as it's faster.

Reply:
I guess it's because there are two operations: subtraction and division. So under hood opencv splits/merges twice.

Reply:
Understood. No need to modify this part.